File: CS50/psets/pset2/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int main(void)
{
    // Prompt the user for text
    string text = get_string("Text: ");

    // Count letters shouldn’t include any punctuation, digits, or other symbols
    // Use islpha() to check if it is a letter
    // Count space to determine number of words
    // Use isblank() to check if it is a space
    // Count punctuations to determine number of sentences
    // Can't use ispunct() to check if it is a punctuation point
    int letter_count = 0;
    int word_count = 1;
    int sentence_count = 0;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        // Store the char to check in a variable
        char char_check = text[i];

        // Count letters, spaces, and punctuations
        if (isalpha(char_check) > 0)
        {
            letter_count++;
        }
        if (isblank(char_check) > 0)
        {
            word_count++;
        }
        if (char_check == '.' || char_check == '!' || char_check == '?')
        {
            sentence_count++;
        }
    }

    // Preview counts
    // printf("%i letter(s)\n", letter_count);
    // printf("%i word(s)\n", word_count);
    // printf("%i sentences(s)\n", sentence_count);

    // Get index = 0.0588 * L - 0.296 * S - 15.8
    float l = (float) letter_count / word_count * 100;
    float s = (float) sentence_count / word_count * 100;
    int index = round(0.0588 * l - 0.296 * s - 15.8);

    if (index < 0)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else if (index > 0 && index < 17)
    {
        printf("Grade %i\n", index);
    }
}

File: CS50/PSets/pset6/readability/readability.py
from cs50.cs50 import get_string


def main():
    # Get text and comput score
    text: str = get_string("Text: ")
    score: int = compute_score(text)

    # Print out put based on score
    print(output(score))


def compute_score(text: str) -> int:
    # Count letters, spaces, and punctuations
    letter_count: int = 0
    word_count: int = 1
    sentence_count: int = 0

    for i in text:
        if i.isalpha():
            letter_count += 1
        if i.isspace():
            word_count += 1
        if i in [".", "!", "?"]:
            sentence_count += 1

    # Calculate L & S
    L: float = float(letter_count / word_count * 100)
    S: float = float(sentence_count / word_count * 100)

    # Calculate the score
    score: float = 0.0588 * L - 0.296 * S - 15.8

    return round(score)


def output(score: int) -> str:
    if score < 1:
        return "Before Grade 1"
    if score > 16:
        return "Grade 16+"
    else:
        return f"Grade {score}"


if __name__ == "__main__":
    main()
